jina/types/sets/document.py

```diff
@@ -155,21 +165,34 @@ def all_contents(self) -> Tuple['np.ndarray', 'DocumentSet']:
         """
         return self._extract_docs('content')
 
-    def _extract_docs(self, attr: str) -> Tuple['np.ndarray', 'DocumentSet']:
-        contents = []
+    def _extract_docs(self, *attr: str) -> Tuple['np.ndarray', 'DocumentSet']:
```

Comment:
Again, I prefer having it more explicit above using `*attr`. I guess, you want to keep it for backward compatibility?

Reply:
backward compatibility and potential use, u would prefer having a List directly?

Reply:
Yes, I'd prefer a list.

Reply:
In this case where passing a single element would be the most common pattern I kind of like this variable length args though, any feelings about this @jina-ai/engineering ?

Reply:
This will become public when we start using batching extensively btw